OpenNefia.Content/GameObjects/EntitySystems/DialogSystem.cs

```diff
@@ -0,0 +1,69 @@
+﻿using OpenNefia.Content.Dialog;
+using OpenNefia.Content.Factions;
+using OpenNefia.Content.Logic;
+using OpenNefia.Content.Prototypes;
+using OpenNefia.Core.GameObjects;
+using OpenNefia.Core.IoC;
+using OpenNefia.Core.Prototypes;
+using OpenNefia.Core.Serialization.Manager.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.GameObjects.EntitySystems
+{
+    public interface IDialogSystem : IEntitySystem
+    {
+        GetDialogueOptionsEventArgs HandleTalk(EntityUid target);
+    }
+    public class DialogSystem : EntitySystem, IDialogSystem
+    {
+        [Dependency] private readonly IFactionSystem _factions = default!;
+        [Dependency] private readonly IMessage _mes = default!;
+
+        private const int TalkDefaultPrio = 20000;
```

Comment:
I'm trying to move away from numeric priority and prefer topological sorting with before/after entries. One example is with the prototype system. In the dialog system, you'd have a string ID like `Elona.TalkDefault`, and when adding new choices you'd specify which IDs should come before/after the new choice.

Reply:
that's a little tricky here, because mods could add choices after the same entry that another mod has already added something after.